components/loggedIn/ProgressCalendar.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  lightFormat,
+  eachDayOfInterval,
+  getWeeksInMonth,
+  startOfWeek,
+  endOfWeek,
+} from "date-fns";
+import { IUserData } from "../../context/user-context";
+import { chunks } from "../../utils/chunks";
+
+interface ISelectedRangeHabits {
+  [key: string]: {
+    [key: string]: boolean;
+  };
+}
+
+const ProgressCalendar = ({
+  habits,
+  checkmarks,
+  selectedHabitKey,
+  selectDatesRange,
+  weekView,
+}: {
+  habits: IUserData["habits"];
+  checkmarks: IUserData["checkmarks"];
+  weekView: boolean;
+  selectedHabitKey: string;
+  selectDatesRange: {
+    start: Date;
+    end: Date;
+  };
+}) => {
+  let habitsKeys = Object.keys(habits);
+  const checkmarkKeys = Object.keys(checkmarks);
+  let selectedDates: string[] = [];
+  let weeksSpanForMonthView: number = getWeeksInMonth(selectDatesRange.start, {
+    weekStartsOn: 1,
+  });
+  let selectedRangeHabits: ISelectedRangeHabits;
+  let selectedDatesChunks: string[][];
+
+  const createSelectedRangeHabitsObject = () => {
```

Comment:
Te funkcje można wynieść do osobnych mniejszych plików i będziesz mogła ich dodatkowo przetestować

Można to zrobić w kolejnym MR